File: TP01/Palindro_Recursivo_Q10&11/Palindromo_recursivo.java
 class Palindromo_recursivo {

   public static boolean isFim(String s){
      return (s.length() >= 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
   }

   public static boolean IsPalindromo (String s, int counter){
      boolean resp = true;

      if(counter >= s.length()){
          resp = true;
      }
      else if(s.charAt(counter) != s.charAt(s.length() - 1 - counter)){
          resp = false;
      }
      else{
         if(counter <= s.length()) resp = IsPalindromo(s,counter+1);
      }

      return resp;
   }

   public static boolean IsPalindromo(String s){
      return IsPalindromo(s, 0);
   }

   public static void main (String[] args){
      String[] entrada = new String[1000];
      int numEntrada = 0;

      //Leitura da entrada padrao
      do {
         entrada[numEntrada] = MyIO.readLine();
      } while (isFim(entrada[numEntrada++]) == false);
      numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

      //Para cada linha de entrada, gerando uma de saida contendo o numero de letras maiusculas da entrada
      for(int i = 0; i < numEntrada; i++){
         MyIO.println(IsPalindromo(entrada[i]) ? "SIM" : "NAO");
      }
   }
}

File: TP01/Palindro_Recursivo_Q10&11/Palindromo_recursivo.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h> 

   bool Fim(char* palavra){
      return (strlen(palavra) >= 3 && palavra[0] == 'F' && palavra[1] == 'I' && palavra[2] == 'M');
   }

   bool IsPalindromo (char* s, int i){
      bool resp;
      if(i >= strlen(s)/2){
         resp = true;
      }
      else if(s[i] != s[strlen(s) - 2 - i]){
        resp = false;
      }
      else{
         resp = IsPalindromo(s,i+1);
      }
        return resp;
   }

   bool chamarPalindromo(char*s){
      return IsPalindromo(s,0);
   }

int main(int argc, char** argv){

    char palavra[1000][1000];
    int numEntrada = 0;
    
   //Ler da entrada padrao as palavras e chamar mestodo para testar se eh a palavra FIM
    do {
       fgets(palavra[numEntrada], 1000, stdin);
    } while (Fim(palavra[numEntrada++]) == false);
    numEntrada--;   
  
   //Chamar metodo de contar letras maisculas e depois exibir na tela a quantidade de letras maisculas.
    for(int i = 0; i < numEntrada; i++){
      chamarPalindromo(palavra[i]) ? printf("SIM\n") : printf("NAO\n");
    }
}